stdlib/public/Platform/android.modulemap

```diff
@@ -0,0 +1,598 @@
+// The module map for the Android NDK.
+
+// The top level 'Android' module can be included from Swift.
+module Android [system] {
+  header "SwiftAndroidNDK.h"
+  export *
+}
+
+// The modules for the Bionic C standard library used by
+// the Android NDK.
+module bionic_assert [system] {
+  // <assert.h>'s use of NDEBUG requires textual inclusion.
+  textual header "assert.h"
+  export *
+}
+module bionic_complex [system] {
+  header "complex.h"
+  export *
+}
+module bionic_ctype [system] {
+  header "ctype.h"
+  export *
+}
+module bionic_errno [system] {
+  header "errno.h"
+  export *
+}
+module bionic_fenv [system] {
+  header "fenv.h"
+  export *
+}
+// Use 'no_undeclared_includes' to avoid pulling in the libc++ headers. 
+module bionic_inttypes [system] [no_undeclared_includes] {
+  header "inttypes.h"
+  export *
+  use bionic_stdint
+  use android_sys_cdefs
+}
+module bionic_limits [system] {
+  header "limits.h"
+  header "bits/posix_limits.h"
+  export *
+}
+module bionic_locale [system] {
+  header "locale.h"
+  export *
+}
+module bionic_malloc [system] {
+  header "malloc.h"
+  export *
+}
+module bionic_math [system] {
+  header "math.h"
+  export *
+  link "m"
+}
+module bionic_pty [system] {
+  header "pty.h"
+  export *
+}
+module bionic_setjmp [system] {
+  header "setjmp.h"
+  export *
+}
+module bionic_signal [system] {
+  header "signal.h"
+  // The 'signal.h' header unfortunately has a circular include
+  // with 'sys/ucontext.h' and thus it must be part of this module.
+  header "sys/ucontext.h"
+  export *
+}
+module bionic_stdatomic [system] {
+  header "stdatomic.h"
+  export *
+}
+module bionic_stdint [system] {
+  header "stdint.h"
+  export *
+}
+module bionic_stdio [system] {
+  header "stdio.h"
+  export *
+}
+module bionic_stdio_ext [system] {
+  header "stdio_ext.h"
+  export *
+}
+module bionic_stdlib [system] {
+  header "stdlib.h"
+  export *
+}
+module bionic_string [system] {
+  header "string.h"
+  export *
+}
+module bionic_time [system] {
+  header "time.h"
+  export *
+}
+module bionic_utmp [system] {
+  header "utmp.h"
+  export *
+}
+
+// POSIX and another android NDK headers.
+module android_alloca [system] {
```

Comment:
Same comment as https://github.com/apple/swift/pull/71840#discussion_r1523803285

I don't know how we want to organize the rest of these. One module per header can start to cause performance issues, but having too many headers in a module can also cause performance issues. Is there some kind of logical grouping we might be able to use like filesystem, networking, etc?

Also it would be nice if we had a common way of naming and grouping these across platforms so that we don't have to do `#if os()` switches to get the same header.

Reply:
Good idea, I can check what we can do. Ideally we would then underscore these groups too.

Reply:
I grouped up some a good amount of individual modules now.